lib/atomic_cache/atomic_cache_client.rb

```diff
@@ -60,13 +59,13 @@ def fetch(keyspace, options={})
 
       # quick check back to see if the other process has finished
       # or fall back to the last known value
-      value = quick_retry(keyspace, options, tags) || last_known_value(keyspace, options, tags)
+      value = quick_retry(key, options, tags) || last_known_value(keyspace, options, tags)
       return value if value.present?
 
       # wait for the other process if a last known value isn't there
       if key.present?
         return time('wait.run', tags: tags) do
-          wait_for_new_value(key, options, tags)
+          wait_for_new_value(keyspace, options, tags)
```

Comment:
`quick_retry` and `wait_for_new_value` changed their approach here a bit:

- `quick_retry` now uses the initial key obtained on line 40, as very little time has passed, and it's more efficient for all the locations that have quick_retry disabled.
- `wait_for_new_value` no longer takes the key, as time will have elapsed between attempts, such that it should make sure it's looking at the most recent key for a value.